### PR TITLE
Avs add new cripto

### PR DESCRIPTION
**Pull request by LuisGerezM**
FIX: fixed accord to comments received in github
App:

https://user-images.githubusercontent.com/8632483/230073095-2982048a-b87c-47d0-88ec-819d63303be0.mp4

------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------

ADD & FIX: redux and persist state added. CryptoList Deleted
App example:

https://user-images.githubusercontent.com/8632483/229853721-4b32adc1-62b1-4b52-b607-e6187c4352a3.mp4

------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------

FIX: fixed crypto list and crypto card in page view. Now are global components"
ADD: added global adapters and all crypto list
-------------------------------------------------------------------------------------------
ADD: form new crypto added

**NOTE: personal crypto list not updated. I need  global state handler. I will make in next feature  **

App example

https://user-images.githubusercontent.com/8632483/229515298-87082900-eb33-4067-a834-d271358fbd1a.mp4

### CODE REVIEW

File: src/components/CryptoCard/index.tsx [hunk 1]
```diff
@@ -19,14 +19,16 @@ interface CryptoCard {
     percent_24: number;
     icon: string;
   };
+  children?: ReactNode;
 }
 
-const CryptoCard = ({crypto}: CryptoCard) => {
+const CryptoCard = ({crypto, children}: CryptoCard) => {
+  // console.log({children}, children ? 'EXISTE' : 'NO EXISTE');
```
Comment (tomasdematos) [inline src/components/CryptoCard/index.tsx]: delete

File: src/components/CryptoCard/styled-components/Card.styled.ts [hunk 2]
```diff
@@ -24,7 +25,9 @@ const WrapPriceAndPercentCrypto = styled.View`
   display: flex;
   flex-flow: column nowrap;
   justify-content: center;
-  align-items: flex-end;
+  align-items: center;
+
```
Comment (tomasdematos) [inline src/components/CryptoCard/styled-components/Card.styled.ts]: delete empty line

File: src/components/WrapCryptoList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+interface WrapCryptoList {
+  children: ReactNode;
+}
+
+const WrapCryptoList = ({children}: WrapCryptoList): JSX.Element => {
+  return <WrapList>{children}</WrapList>;
+};
+export default WrapCryptoList;
+
+const WrapList = styled.ScrollView`
```
Comment (tomasdematos) [inline src/components/WrapCryptoList/index.tsx]: styles should be stored in a different file

File: src/components/WrapCryptoList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React, {ReactNode} from 'react';
+import styled from 'styled-components/native';
+
+interface WrapCryptoList {
+  children: ReactNode;
+}
+
+const WrapCryptoList = ({children}: WrapCryptoList): JSX.Element => {
```
Comment (tomasdematos) [inline src/components/WrapCryptoList/index.tsx]: use implicit return

File: src/hooks/useAlertUserConfirm.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  title: string;
+  message: string;
+}
+
+export const useAlertUserConfirm = () => {
+  const showAlertUserConfirm = async ({title, message}: AlertUserConfirm) =>
+    new Promise(resolve => {
+      Alert.alert(title, message, [
+        {
+          text: 'aceptar',
```
Comment (tomasdematos) [inline src/hooks/useAlertUserConfirm.ts]: is in spanish

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+            message:
+              'Search not valid. Search for name or symbol, e.g.: Bitcoin or BTC',
+          });
+          return;
+        }
+
+        const {id, name, symbol, percent_24, price_usd} = fetching;
+
+        const message = createCryptoFoundMsg(
+          'Moneda encontrada',
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: in english

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+          name,
+          symbol,
+          price_usd,
+          percent_24,
+        );
+
+        const userConfirm = await showAlertUserConfirm({
+          title: 'Busqueda con éxito',
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: all should be in english, check all the texts

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        if (userConfirm) {
+          setAddNewCrypto({id});
+          showAlertUserFeedback({
+            title: 'Operation successfully',
+            message: `Crypto ${name} added correctly`,
+          });
+          resetField('criptoCurrency');
+        }
+      } catch (error) {
+        console.error('Error searching cripto', error.message);
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: delete console.log

File: src/pages/AddCryptoCurrency/index.tsx [hunk 9]
```diff
@@ -1,7 +1,25 @@
...
+import {Form} from './components';
+import AllCryptoList from './components/AllCryptosList';
+import {
+  WrapAddCripto,
+  WrapButtonBack,
+} from './styled-components/AddCriptoCurrency.styled';
 
-const AddCryptoCurrency = (): JSX.Element => {
-  return <Text>AddCryptoCurrency</Text>;
+const AddCriptoCurrency = (): JSX.Element => {
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/index.tsx]: use implicit return

File: src/components/CryptoCard/index.tsx (hunk 1 above)
Comment (LuisGerezM) [reply src/components/CryptoCard/index.tsx]: Ups.. sorry

File: src/components/CryptoCard/styled-components/Card.styled.ts (hunk 2 above)
Comment (LuisGerezM) [reply src/components/CryptoCard/styled-components/Card.styled.ts]: Okey.

File: src/components/WrapCryptoList/index.tsx (hunk 3 above)
Comment (LuisGerezM) [reply src/components/WrapCryptoList/index.tsx]: Okey.

File: src/components/WrapCryptoList/index.tsx (hunk 4 above)
Comment (LuisGerezM) [reply src/components/WrapCryptoList/index.tsx]: Okey

File: src/hooks/useAlertUserConfirm.ts (hunk 5 above)
Comment (LuisGerezM) [reply src/hooks/useAlertUserConfirm.ts]: Ups... sorry

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts (hunk 7 above)
Comment (LuisGerezM) [reply src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: Okey.

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts (hunk 8 above)
Comment (LuisGerezM) [reply src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: that "console.error" is for the programmer detect where it failed so it is error and not log, but I remove it, no problem.

File: src/pages/AddCryptoCurrency/index.tsx (hunk 9 above)
Comment (LuisGerezM) [reply src/pages/AddCryptoCurrency/index.tsx]: Okey

File: src/App.tsx [hunk 10]
```diff
@@ -8,39 +8,57 @@
...
+import {Header, SpinnerLoader} from './components';
 import {routes} from './models';
-import {Header} from './components';
 import {AddCryptoCurrency, Home} from './pages';
+import {store} from './redux/store';
 import {defaultTheme} from './styled-components/theme/theme.styled';
 
 const Stack = createNativeStackNavigator();
 
+let persistor = persistStore(store);
```
Comment (tomasdematos) [inline src/App.tsx]: use const

File: src/pages/AddCryptoCurrency/components/Form/index.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+      )}
+
+      <WrapAddButton isLoading={loadingSearchCripto}>
+        {loadingSearchCripto && <Text>Looking... </Text>}
+        <Button
+          onPress={handleSubmit(onSubmit)}
+          backgroundColor="yellow"
+          borderColor="grey"
+          width="50%"
+          disabled={!!loadingSearchCripto}
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/components/Form/index.tsx]: you should validate if the input is empty

File: src/components/Errors/ErrorsValidationForm/ErrorsValidationForm.tsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  [key: string]: any;
+  errorKey: string;
+}
+
+export const ErrorsValidationForm = ({
+  errors,
+  errorKey,
+}: ErrorsValForm): JSX.Element => {
+  return errors[errorKey] ? (
+    <TextComponent fontSizeType="text" fontWeight="bold" color="red">
```
Comment (tomasdematos) [inline src/components/Errors/ErrorsValidationForm/ErrorsValidationForm.tsx:14]: use theme in all the cases where you use a color

File: src/pages/AddCryptoCurrency/components/Form/index.tsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+  return (
+    <WrapForm>
+      <TextComponent fontSizeType="title" fontWeight="heavy" color="darkBlue">
+        Add a Cryptocurrency
+      </TextComponent>
+      <Controller
+        control={control}
+        rules={validationFields.criptoCurrency}
+        render={({field: {onChange, onBlur, value}}) => (
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/components/Form/index.tsx:32]: move to a render function

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+
+  const dispatch = useDispatch();
+  const cryptoData = useSelector(state => state.cryptoData);
+
+  const navigation = useNavigation();
+
+  const [areSearching, setAreSearching] = useState<string>('');
+
+  const [loadingSearchCripto, setLoadingSearchCripto] =
+    useState<boolean>(false);
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: unnecessary type declaration, is already implicit

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+      criptoCurrency: '',
+    },
+  });
+
+  const dispatch = useDispatch();
+  const cryptoData = useSelector(state => state.cryptoData);
+
+  const navigation = useNavigation();
+
+  const [areSearching, setAreSearching] = useState<string>('');
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts]: unnecessary type declaration, is already implicit

File: src/pages/AddCryptoCurrency/components/Form/index.tsx (hunk 11 above)
Comment (tomasdematos) [reply src/pages/AddCryptoCurrency/components/Form/index.tsx]: also, 
if loadingSearchCripto is true, !!true is equal to true, 
just use the loadingSearchCripto alone

Comment (tomasdematos) [reply src/pages/AddCryptoCurrency/components/Form/index.tsx]: also if it is undefined or null this two cases are both false by default in js

File: src/components/Errors/ErrorsValidationForm/ErrorsValidationForm.tsx (hunk 12 above)
Comment (LuisGerezM) [reply src/components/Errors/ErrorsValidationForm/ErrorsValidationForm.tsx:14]: I'm using theme in TextComponent writed in GlobalStyles.styled.ts
```
const TextComponent = styled.Text<FontProps>`
  font-size: ${({fontSizeType, theme}: FontProps) =>
    theme.fontSizeType[fontSizeType]};
  font-weight: ${({fontWeight, theme}: FontProps) =>
    theme.fontWeights[fontWeight]};
  color: ${({color, theme}: FontProps) => theme.colors[color]};
  text-align: ${props => props.textAlign || 'center'};
`;
```

Comment (tomasdematos) [reply src/components/Errors/ErrorsValidationForm/ErrorsValidationForm.tsx:14]: ok

File: src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+        }
+      } catch (error) {
+        errorType(error.message, showAlertUserFeedback);
+        setAreSearching('');
+      } finally {
+        setLoadingSearchCripto(false);
+      }
+    };
+
+    if (areSearching) {
```
Comment (tomasdematos) [inline src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts:118]: it shoud be: isSearching

Comment (LuisGerezM) [reply src/pages/AddCryptoCurrency/hooks/useAddCryptoForm.ts:118]: Yes.!

**Pull request merged**